### PR TITLE
added PATCH reviews and error handling

### PR DESCRIPTION
**Pull request by Chris9533**

### CODE REVIEW

File: Controllers/reviews.controllers.js [hunk 1]
```diff
@@ -9,4 +9,23 @@ res.status(200).send({ review })
...
+}
+
+exports.patchReview = (req, res, next) => {
+    const review_id = req.params.review_id
+    let alterVotes = req.body.inc_votes  
+
+    fetchReview(review_id).then((review) => {
+        let votes = review[0].votes
+        let newVotes = votes += alterVotes
+
```
Comment (bainesface) [inline Controllers/reviews.controllers.js:21]: Hey Chris!

How come you are fetching the review here before updating? I would say all the behaviour required can be handled in the one updateReview model function!

File: Models/reviews.models.js [hunk 2]
```diff
@@ -11,5 +11,14 @@ exports.fetchReview = (review_id) => {
        return review;
       }
 
+    })
+}
+
+exports.updateReview = (newVotes, review_id) => {
+    return db.query("UPDATE reviews SET votes = $1 WHERE review_id = $2 RETURNING *", [newVotes, review_id] )
+    .then((result) => {
```
Comment (bainesface) [inline Models/reviews.models.js:19]: Have a look at whether we can set the votes with some sort of additional operation

File: __tests__/games.test.js [hunk 3]
```diff
@@ -98,4 +98,78 @@ afterAll(() => {
...
+
+  describe.only('PATCH /api/reviews/review_id', () => {
+      test('202: responds with a review with it\'s votes counter altered by the correct amount', () => {
+          const inc_votes = { inc_votes: -5 }
+
+          return request(app)
+          .patch("/api/reviews/2")
+          .send(inc_votes)
+          .expect(202)
+          .then((results) => {
```
Comment (bainesface) [inline __tests__/games.test.js:111]: Whilst I like your use of a 202 status code, this means that a request has been accepted but hasn't yet been processed. This isn't accurate in this case sou I would use a 200 instead

File: __tests__/games.test.js [hunk 4]
```diff
@@ -98,4 +98,78 @@ afterAll(() => {
...
+            const review = results.body.review
+            
+            expect(review.length).toBe(1)
+
```
Comment (bainesface) [inline __tests__/games.test.js:115]: As we are sending back a single review I would take the review object out of the array before sending as a response. It's convention to put multiples in an array but send a single as an object

File: __tests__/games.test.js [hunk 5]
```diff
@@ -98,4 +98,78 @@ afterAll(() => {
...
+        .expect(400)
+        .then(({body}) => {
+
+            expect(body.msg).toBe("bad request")
+          
+      });
+        
+    });
+
+
```
Comment (bainesface) [inline __tests__/games.test.js:174]: One final test here. What if the request body you send across does not contain a key of `inc_votes`?

File: __tests__/games.test.js [hunk 6]
```diff
@@ -98,4 +92,90 @@ afterAll(() => {
           
       });
       
+  });
+
+  describe('PATCH /api/reviews/review_id', () => {
+      test('202: responds with a review with it\'s votes counter altered by the correct amount', () => {
```
Comment (Augs0) [inline __tests__/games.test.js:98]: don't forget to update the test text to reflect the status code you're using now

**Review by Augs0 — APPROVED** [review]

**Pull request merged**